scripts/hack-on

```diff
@@ -48,8 +48,18 @@ if [ -f "$REPO/thunk.nix" ] ; then
     rm "$REPO/thunk.nix"
 fi
 rm "$JSON_FILE"
+
+# Install the cleanup trap handler *after* `rmdir` to avoid overwriting any
+# newly added files that are preventing the directory from being removed.
+trap "echo 'Attempted to replace $REPO but encountered unexpected file(s). Please remove.'" ERR
 rmdir "$REPO"
 
+# Though it might be best to use some combination of `checkout`, `clean`, and
+# `restore`, this approach is most widely supported across git versions.
+# Additionally, this is also the least surprising when dealing with oddities
+# around nested git repositories at the price of a scary `rm -f`.
+trap "rm -rf $REPO && git checkout $REPO 2> /dev/null" ERR
```

Comment:
While this is a good intention, at this point in time I would prefer that we do not try to do this. If `hack-on` fails, that is an unusual situation and I would prefer to do as little as possible when the user is in an unusual error condition. Instead, let us notify the user that there is was an issue and that the checkout of this thunk may be in a bad state.